[#846] Docs for airy config yaml (#873)
airyhq/airy: docs/docs/getting-started/deployment/configuration.md

---
title: Configuration
sidebar_label: Configuration
---

## Airy Core Configuration File

> The `infrastructure/airy.tpl.yaml` file contains examples of all possible configuration options. Copy and rename it to `infrastructure/airy.yaml` and modify it according to your environment:

Reviewer: ```suggestion
The `infrastructure/airy.tpl.yaml` file contains examples of all possible configuration options. Create your own `airy.yaml` like so:
```

Author: love it!

```bash
cd infrastructure
cp airy.tpl.yaml airy.yaml
```

> We will guide you through the different sections so you can make the changes you are looking for. Keys can also be omited for services which you do not wish to configure.

Reviewer: ```suggestion
We will guide you through the different sections so you can make the changes you are looking for. Keys can also be omitted for services which you do not wish to configure.
```

### Global

- **appImageTag**

>   image tag of the container images for the `Airy Components`

Reviewer: We should probably specific what kind of value they can expect here (as in the "official released" tags plus develop)

- **containerRegistry**

>   The images of the `Airy Components` are stored in the [Github Container Registry](https://docs.github.com/en/packages/guides/about-github-container-registry).

Reviewer: The subject here should be something like "The URL of the container registry" I think (unless I'm missing something)

Author: I just don't want people to encourage changing it but yeah it's their fault if they do so I'll change it

- **namespace**

>   Kubernetes namespace the `Airy Core` will use

Reviewer: ```suggestion
  the Kubernetes namespace the `Airy Core` will use
```

### Prerequisites

These settings are used to connect the `Airy Components` to your Kafka cluster, PostgreSQL and Redis.

- **kafka**

  `commitInterval` sets the [Kafka Commit Interval](https://kafka.apache.org/documentation/#consumerconfigs_auto.commit.interval.ms) if you are using the included Helm chart.

- **redis**
- **postgres**

  We recommend that you create a new database if you are reusing a PostgreSQL server to avoid name collisions.

### Components

- **api**

>   `mail*` settings are needed if you want the Airy Core to send emails out

Reviewer: as there's no "public feature" using this (and I believe... we will actually never have one as there's better ways to achieve this integration), I would remove this one

>   `jwtSecret` should be set to a long secure secret in production environments

Reviewer: this is the HTTP api JWT signing key right? I think the name has to be more explicit (I get we'd have to change the code... but if not now when? :))

  `allowedOrigins` set this to your sites origin

- **sources**

>   The `Airy Controller` will only start `Airy Components` when they are configured here. So to keep system load to a minimum only add the sources you are using.

Reviewer: ```suggestion
  The `Airy Controller` only starts configured sources. To keep system load to a minimum, only add the sources you are using.
```

- **webhooks**
- **media-resolver**

## Applying the configuration

If you made changes in `airy.yaml` and want to apply it to your instance you can use the [CLI](getting-started/cli.md) by running the following command.

```bash
airy config apply --config ./airy.yaml --kube-config /path/to/your/kube.conf
```

> The Airy CLI considers that the Kubernetes configuration file is located under

Reviewer: this info belongs to the CLI docs I think (which also makes me think we should follow up with a ticket and have a `--config-dir` option). I will create the ticket!

Author: ok I'll link it here instead then

  - `hostname`
  - `port`

  Redis is needed as a queue for the [Webhooks](/api/webhook.md)

- `postgres`
  - `endpoint` in `<HOSTNAME>:<PORT>` format
  - `dbName` name of the database in the PostgreSQL server
  - `username` these credentials will be passed to the **API Auth Component**
  - `password` and they will be used to create the Postgres if you are deploying with **Vagrant**

### Components

- `api`

  - `jwtSecret` should be set to a long secure secret in production environments
  - `allowedOrigins` your sites origin to prevent CORS-based attacks

- `sources`

  - `facebook`
  - `google`
  - `twilio`

  The **Airy Controller** only starts configured sources. To keep system load to a minimum, only add the sources you are using.

- `webhooks`
  - `name`
- `media-resolver`
  - `storage`
    - `s3` set these to your AWS S3 config to store source specific user data

## Applying the configuration

If you made changes in `airy.yaml` and want to apply it to your instance you can use the [airy config apply](/cli/airy_config_apply.md) by running the following [CLI](getting-started/cli.md) command.

```bash
airy config apply --config ./airy.yaml --kube-config /path/to/your/kube.conf

> ```

Reviewer 2: I think that the content of the document is good, but this last command, I would put it inside the `Vagrant` and `Run-in-production` docs, so that all the commands which one needs to run are in a single document.
